app/indexers/queued_nesting_indexer.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class QueuedNestingIndexer
+  extend Samvera::NestingIndexer
+  def self.reindex_relationships(id:, maximum_nesting_depth: nil, extent:)
+    if extent.match("queue")
+      Rails.logger.info("nested indexing queued")
+      Redis.current.zadd("nested:index:#{extent.delete("queue")}", 0, id.to_s)
```

Comment:
importing_id is the id of the Importer, not of the individual Import Entry.  So for "Test Import 18", the id might be 40. We queue all the records from that import in to "nested:index:40" and then pull them IndexAfterJob in sets of 100 for batch conversion to Solr. By indexing 100 records at a time in Solr instead of 1 record at a time, we cut the number of http requests for Solr and make indexing more efficient ON TOP of the fact that we solved the N+1 indexing issue by queuing them in the first place. This implementation was informed by Searchkick, and Ruby gem for Elastic Search which has a similar queued indexing method.

Reply:
👍